src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultProjectConfigurationFilePathStore.cs

```diff
@@ -15,25 +15,25 @@ namespace Microsoft.VisualStudio.LanguageServerClient.Razor;
 [Export(typeof(ProjectConfigurationFilePathStore))]
 internal class DefaultProjectConfigurationFilePathStore : ProjectConfigurationFilePathStore
 {
-    private readonly Dictionary<string, string> _mappings;
+    private readonly Dictionary<ProjectKey, string> _mappings;
     private readonly object _mappingsLock;
 
     public override event EventHandler<ProjectConfigurationFilePathChangedEventArgs>? Changed;
 
     [ImportingConstructor]
     public DefaultProjectConfigurationFilePathStore()
     {
-        _mappings = new Dictionary<string, string>(FilePathComparer.Instance);
+        _mappings = new Dictionary<ProjectKey, string>();
         _mappingsLock = new object();
     }
 
-    public override IReadOnlyDictionary<string, string> GetMappings() => new Dictionary<string, string>(_mappings);
+    public override IReadOnlyDictionary<ProjectKey, string> GetMappings() => new Dictionary<ProjectKey, string>(_mappings);
```

Comment:
Can we make this `ImmutableDictionary<ProjectKey, string>`? Now seems like a good time

Reply:
We can. I'm not sure how much to do in these PRs versus followups, to keep them easier to review. Certainly for me, I like that I can scroll down a diff and see just "string changed to ProjectKey" pretty quickly.